formula-lint/src/main/java/com/instacart/formula/lint/WrongFormulaUsageDetector.kt

```diff
@@ -39,27 +42,60 @@ class WrongFormulaUsageDetector : Detector(), Detector.UastScanner {
             )
         )
 
+        val ISSUE_KEYLESS_CALLBACKS_WITHIN_LOOP = Issue.create(
+            id = "KeylessFormulaCallbackWithinLoop",
+            briefDescription = "Missing key in Formula callback.",
+            explanation = "It is an error to not provide an unique [key] for a FormulaContext#callback call within a loop or Iterable function. This will result in a Formula runtime crash in most of the cases.",
+            category = CORRECTNESS,
+            priority = 9,
+            severity = ERROR,
+            implementation = Implementation(
+                WrongFormulaUsageDetector::class.java,
+                EnumSet.of(Scope.ALL_JAVA_FILES)
+            )
+        )
+
         val issues = arrayOf(
             ISSUE_ILLEGAL_CALL_WITHIN_TRANSITION_CONTEXT,
+            ISSUE_KEYLESS_CALLBACKS_WITHIN_LOOP,
         )
     }
 
-    override fun getApplicableUastTypes(): List<Class<out UElement>>? {
+    override fun getApplicableUastTypes(): List<Class<out UElement>> {
         return listOf(
             UExpression::class.java,
         )
     }
+
     sealed class FormulaReference(val name: String) {
         object Snapshot : FormulaReference("Snapshot")
         object FormulaContext : FormulaReference("FormulaContext")
     }
 
-    override fun createUastHandler(context: JavaContext): UElementHandler? {
-        return object: UElementHandler() {
+    override fun createUastHandler(context: JavaContext): UElementHandler {
+        return object : UElementHandler() {
             override fun visitExpression(node: UExpression) {
                 if (node is UCallExpression) {
                     val resolved = node.resolve()
                     val methodOwner = getMethodOwner(context, resolved)
+
+                    if (methodOwner != null && isKeylessFormulaCallback(context, node) && isWithinKeylessLoop(context, node)) {
+                        val call = node.sourcePsi
+                        context.report(
+                            Incident(
+                                issue = ISSUE_KEYLESS_CALLBACKS_WITHIN_LOOP,
+                                scope = call,
+                                location = context.getLocation(call),
+                                message = """
+                                    |Key-less context.callback() call within an Iterable. 
+                                    |This will result in a runtime crash for a loop with more than 1 iteration. 
```

Comment:
After auto-indexing was added, it won't result in runtime crash any more. However, it's still better to have a stable key instead of index only. Using a stable unique key allows us to re-use same callback across re-evaluation which enables proper equality checks.

Reply:
Got it. I will change the copy here. I guess the fact that it's not crashing anymore - also impacts this issue severity (`ERROR`) and priority (`9`)?

Reply:
Likely still would keep it as an error